src/Integration/LocalServices/TestProjectIndicators/BuildPropertyTestProjectIndicator.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration.Resources;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal class BuildPropertyTestProjectIndicator : ITestProjectIndicator
```

Comment:
@duncanp-sonar I'm thinking about ditching this class, and moving all of its logic into TestProjectIndicator.
1. This way the interface can be bool and not bool?
2. TestProjectIndicator is highly aware of this anyway due to precedence, so maybe we should remove this separation.

Reply:
I think it's ok to have both; this class encapsulates knowledge about accessing build properties, `TestProjectIndictor` encapsulates the fact that there is a 'primary' indicator that should take precedence over secondary indicators. 

`TestProjectIndicator' doesn't need to know which indicators are primary and which are secondary, just that there is a precedence rule and how to apply it.